server/controllers/post-controller.ts

```diff
@@ -0,0 +1,103 @@
+import express from 'express';
+import Boom from '@hapi/boom';
+import { Request, Response, NextFunction, RequestHandler } from 'express-serve-static-core';
+import { Dependencies } from '../types';
+import { PostService } from '../service/post-service';
+import { UserService } from '../service/user-service';
+
+export class PostController {
+  notFound: Boom;
+  router: express.Router;
+  postService: PostService;
+  userService: UserService;
+  validatePost: RequestHandler;
+  validatePostPUT: RequestHandler;
+  validateParams: RequestHandler;
+  constructor(options: Dependencies) {
+    this.router = express.Router();
+    this.notFound = Boom.notFound("Post doesn't exist");
+    this.postService = options.postService;
+    this.userService = options.userService;
+    this.validatePost = options.validatePost;
+    this.validatePostPUT = options.validatePostPUT;
+    this.validateParams = options.validateParams;
+    this.initializeRoutes();
+  }
+
+  initializeRoutes() {
+    this.router.get('/', this.getAll.bind(this));
+    this.router.get('/:id', this.validateParams, this.getOne.bind(this));
+    this.router.post('/', this.validatePost, this.post.bind(this));
+    this.router.delete('/:id', this.validateParams, this.delete.bind(this));
+    this.router.put('/:id', this.validateParams, this.validatePostPUT, this.update.bind(this));
+  }
+
+  async getAll(req: Request, res: Response, next: NextFunction) {
+    try {
+      const posts = await this.postService.find();
+      res.status(200).send(posts);
+    } catch (error) {
+      next(Boom.badImplementation());
+    }
+  }
+
+  async getOne(req: Request, res: Response, next: NextFunction) {
+    try {
+      const post = await this.postService.findOne(req.params.id);
+      if (post) {
+        res.status(200).send(post);
+      } else {
+        next(this.notFound);
+      }
+    } catch (error) {
+      next(Boom.badImplementation());
+    }
+  }
+
+  // Create post
+  async post(req: Request, res: Response, next: NextFunction) {
```

Comment:
Would `create` be better name for this function? In general if there has to be comment to clarify the function then the function is badly named :P

Reply:
At the time it was serving under the name of router, it seemed suitable for the methods to be named after the HTTP Requests! I agree with the renaming though! Switched gets to reads as well!